### ISSUE TITLE
Mysterious boundary for numeral recognization

### ISSUE THREAD

**Issue by SaltfishAmi** [birchill/10ten-ja-reader#1485]
![image](https://github.com/birchill/10ten-ja-reader/assets/14184974/bf84b8b9-1b1a-4928-8e95-0a0d63684be5)

I even don't know how to describe this issue as it's kind of hilarious...

I put my cursor at the first `八` and apparently 10ten recognized the two kanji `八`s and the first digit `2` in the date as a number `882`.  For some reason it did not stop at the emoticon `img` and give me `88` nor continue going on and give me `882023`.

**Comment by birtles** [comment]
Are you on version 1.16? When I go there (on Nightly) I get:

![image](https://github.com/birchill/10ten-ja-reader/assets/1232595/f94be641-0097-44bc-8586-87cd162a49e3)

**Comment by SaltfishAmi** [comment]
@birtles OK, I confirmed same behavior as in your picture on 1.16.

Still, recognizing different numerals (and having an image inbetween) as one number seems like a mystery to me.

**Comment by birtles** [comment]
Yeah, I agree. I'm not sure if there's a reason we allow mixing numerals. We need to at least allow mixing Arabic numerals with kanji powers of ten so we can recognize 200万 etc. but I don't know if there's ever a case for mixing kanji 一～九 with Arabic numerals.

I'm also not sure if we need to be able to skip over images or whether they should always be treated as word boundaries.

**Comment by willbonney** [comment]
Is this the same issue? Recognizing un-suffixed arabic numerals on their own?

<img width="845" height="192" alt="Image" src="https://github.com/user-attachments/assets/e6c3bee9-2d11-4901-8988-e2b4636cdd64" />

**Comment by birtles** [comment]
> Is this the same issue? Recognizing un-suffixed arabic numerals on their own?

I don't think it's quite the same issue. It's curious we match 64 but not ６４ (full-width). I'd expect us to do the opposite.
